File: api/src/services/vk.ts
import type { UserInfoResult } from "@vkid/sdk";

export class VkService {
  async getUserInfo(access_token: string): Promise<UserInfoResult> {
    const response = await fetch(
      `https://${process.env.VKID_DOMAIN}/oauth2/user_info`,
      {
        method: "POST",
        body: new URLSearchParams({
          client_id: process.env.VK_CLIENT_ID,
          access_token,
        }),
      }
    );

    const data = await response.json();

    return data;
  }
}

export const vkSvc = new VkService();

File: api/oauth/vk.py
from settings.settings import Settings
import requests


def get_user_info(token: str):
    params = {
        "client_id": Settings.auth.vk_client_id,
        "access_token": token,
    }
    headers = {'Content-Type': 'application/x-www-form-urlencoded'}
    r = requests.post("https://id.vk.com/oauth2/user_info",
                      data=params, headers=headers)
    return r.json()
